pymatgen/io/vasp/inputs.py

```diff
@@ -1773,55 +1756,45 @@ def __init__(self, data, symbol=None):
+        self.keywords = dict(sorted({**PSCTR, **keywords}.items()))
 
         if symbol:
             self._symbol = symbol
         else:
             try:
-                self._symbol = self.keywords["TITEL"].split(" ")[1].strip()
+                self._symbol = keywords["TITEL"].split(" ")[1].strip()
             except IndexError:
-                self._symbol = self.keywords["TITEL"].strip()
+                self._symbol = keywords["TITEL"].strip()
 
-        # Compute the POTCAR hashes to check them against the database of known
-        # VASP POTCARs and possibly SHA256 hashes contained in the file itself.
-        self.hash = self.get_potcar_hash()
-        self.file_hash = self.get_potcar_file_hash()
-        if hasattr(self, "SHA256"):
-            self.hash_sha256_from_file = self.SHA256.split()[0]
-            self.hash_sha256_computed = self.get_sha256_file_hash()
+        # Compute the POTCAR meta to check them against the database of known metadata,
+        # and possibly SHA256 hashes contained in the file itself.
 
-        if not self.identify_potcar(mode="data")[0]:
-            warnings.warn(
-                f"POTCAR with symbol {self.symbol} has metadata that does\n"
-                "not match any VASP POTCAR known to pymatgen. The data in this\n"
-                "POTCAR is known to match the following functionals:\n"
-                f"{self.identify_potcar(mode='data')[0]}",
-                UnknownPotcarWarning,
-            )
+        sha256_pass = True
+        if has_sha256 := hasattr(self, "SHA256"):
+            self.hash_sha256_from_file = self.SHA256.split()[0]
+            self.hash_sha256_computed = self.sha256_file_hash
+            sha256_pass = self.hash_sha256_from_file == self.hash_sha256_computed
 
-        has_sh256, hash_check_passed = self.verify_potcar()
-        if not has_sh256 and not hash_check_passed:
+        if not has_sha256 and not self.is_valid:
             warnings.warn(
-                f"POTCAR data with symbol {self.symbol} does not match any VASP "
-                "POTCAR known to pymatgen. There is a possibility your "
-                "POTCAR is corrupted or that the pymatgen database is incomplete.",
+                f"POTCAR data with symbol {self.symbol} is not known to pymatgen. Your "
+                "POTCAR may be corrupted or pymatgen's POTCAR database is incomplete.",
                 UnknownPotcarWarning,
             )
-        elif has_sh256 and not hash_check_passed:
+        elif has_sha256 and not sha256_pass:
             warnings.warn(
                 f"POTCAR with symbol {self.symbol} and functional\n"
                 f"{self.functional} has a SHA256 hash defined,\n"
                 "but the computed hash differs.\n"
-                "YOUR POTCAR FILE HAS BEEN CORRUPTED AND SHOULD NOT BE USED!"
+                "YOUR POTCAR FILE HAS BEEN CORRUPTED AND SHOULD NOT BE USED!",
+                UnknownPotcarWarning,
```

Comment:
@esoteric-ephemera @mkhorton @Andrew-S-Rosen Are we sure we want to keep the old unknown POTCAR warning when the SHA256 hash provided by VASP doesn't match the one computed from the file? Isn't that what has been giving so many false positive warnings?

Reply:
Does this PR not make this more robust, reducing false positives?

The warning is definitely badly phrased if we do keep it, it’s inscrutable unless someone is very familiar with VASP and pymatgen. We could link to a docs page for explanation.

Reply:
> Does this PR not make this more robust, reducing false positives?

The new warning that's supposed to be more robust comes just above these lines:

https://github.com/materialsproject/pymatgen/blob/c329ff56e65be1c4f77d5e5a6de168e683902133/pymatgen/io/vasp/inputs.py#L1778-L1783

Currently, when `not self.is_valid` is `False`, i.e. the POTCAR is valid according to the new validation, then we still check whether computed and in-file hash match and trigger the old warning if not (which has many false positives, I think):
```
elif has_sha256 and not sha256_pass:
    warnings.warn(
                f"POTCAR with symbol {self.symbol} and functional\n"
                ...
```

Reply:
The SHA56 warning is not what's been giving us so many `UnknownPotcarWarnings`, that was the internal PMG hashing which has been removed. Some versions of VASP POTCARs have a SHA56 tag in the header that VASP computed on their side. We don't have any SHA56-tagged POTCARs for me to test that the current validator works well in their place

Reply:
Though honestly, given the comments on issue [#2895](https://github.com/materialsproject/pymatgen/issues/2895), it seems like the SHA56 has similar fragility as PMG's hashes. I would lean towards removing the SHA56 check

Reply:
> given the comments on issue https://github.com/materialsproject/pymatgen/issues/2895, it seems like the SHA56 has similar fragility as PMG's hashes

that was my understanding as well

>  I would lean towards removing the SHA56 check

same!